File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {

  let string=[]
  for(let i = str.length-1; i >=0;i--){
    string.push(str[i])
  }
 let revisedStr=string.join('')
 return revisedStr

}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

//split the word 
//push it to an array
//revrese array
//join as a string and retrn

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
  string=[]
 
  i=str.length-1

  while i >= 0
    string.push(str[i])
    i -= 1
  end

  string.join
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file

# And a written explanation of your solution

#split the word 
#push it to an array
#revrese array
#join as a string and retrn